src/layout/Payment/config.ts

```diff
@@ -0,0 +1,54 @@
+import { CG } from 'src/codegen/CG';
+import { ExprVal } from 'src/features/expressions/types';
+import { CompCategory } from 'src/layout/common';
+
+export const Config = new CG.component({
+  category: CompCategory.Presentation,
+  rendersWithLabel: false,
+  capabilities: {
+    renderInTable: false,
+    renderInButtonGroup: false,
+    renderInAccordion: false,
+    renderInAccordionGroup: false,
+  },
+})
+  .addProperty(
+    new CG.prop(
+      'renderAsSummary',
+      new CG.expr(ExprVal.Boolean)
+        .optional({ default: false })
+        .setTitle('Render as summary')
+        .setDescription(
+          'Boolean value or expression indicating if the component should be rendered as a summary. Defaults to false.',
+        ),
+    ),
+  )
+  .addTextResource(
+    new CG.trb({
+      name: 'title',
+      title: 'Title',
+      description: 'The title of the paragraph',
+    }),
+  )
+  .addTextResource(
+    new CG.trb({
+      name: 'description',
+      title: 'Description',
+      description: 'Description, optionally shown below the title',
+    }),
+  )
+  .addProperty(
+    new CG.prop(
+      'paymentSettings',
```

Comment:
Settings on the Payment component are, by their definition, 'payment settings' - why the need to require an object here? I would just have this one setting on the root of the Payment component.